lib/autoupdate/agent/process.go

```diff
@@ -0,0 +1,179 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package agent
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"log/slog"
+	"os"
+	"os/exec"
+
+	"github.com/gravitational/trace"
+)
+
+// SystemdService manages a Teleport systemd service.
+type SystemdService struct {
```

Comment:
nit: btw Teleport has indirect dependency of https://github.com/coreos/go-systemd which we might use https://github.com/coreos/go-systemd/blob/main/dbus/methods_test.go#L205
I don't say that is better, just suggestion

Reply:
Thanks for pointing this out. I lean towards using systemctl for now, since it's guaranteed to be present in the context the agent is executing on a machine with systemd, and we're likely going to port the existing updater logic for now. We can revisit later.